src/testing/deploy/deploy_test.go

```diff
@@ -34,6 +34,28 @@ func TestDeployEnterprise(t *testing.T) {
 	mockIDPLogin(t, c)
 }
 
+func TestUpgradeEnterpriseWithEnv(t *testing.T) {
+	k := testutil.GetKubeClient(t)
+	opts := &minikubetestenv.DeployOpts{
+		AuthUser:     auth.RootUser,
+		Enterprise:   true,
+		CleanupAfter: true,
+	}
+	c := minikubetestenv.InstallRelease(t, context.Background(), "default", k, opts)
+	whoami, err := c.AuthAPIClient.WhoAmI(c.Ctx(), &auth.WhoAmIRequest{})
+	require.NoError(t, err)
+	require.Equal(t, auth.RootUser, whoami.Username)
+	// set new root token via env
+	opts.AuthUser = ""
+	token := "new-root-token"
+	opts.ValueOverrides = map[string]string{"pachd.rootToken": token}
+	c = minikubetestenv.UpgradeRelease(t, context.Background(), "default", k, opts)
+	c.SetAuthToken(token)
+	whoami, err = c.AuthAPIClient.WhoAmI(c.Ctx(), &auth.WhoAmIRequest{})
+	require.NoError(t, err)
+	require.Equal(t, auth.RootUser, whoami.Username)
```

Comment:
Can we also check that the old token doesn't somehow still work?

(Some random complexity that you don't have to deal with here, but we might want to think about in the future.  I like non-aborting assertions for these test cases where we want to test two things, so that when it fails, it will say "old key didn't work\n new key didn't work".  That gives you more information about the state of the system when the test fails. 

 I don't actually know if we have non-aborting assertions in our custom assertion library, though.)

Reply:
we don't have the non-aborting assertions, but I can add the check :)